--- src/modules/gui/mod.rs
+++ src/modules/gui/mod.rs
@@ -445,3 +445,5 @@
             Message::EditorAction(action) => {
                 self.editor_content.perform(action.clone());
-                self.is_dirty = true;
+                if let cosmic::widget::text_editor::Action::Edit(_) = action {
+                    self.is_dirty = true;
+                }
             }
--- src/modules/gui/view.rs
+++ src/modules/gui/view.rs
@@ -216,4 +216,4 @@
         let btn = cosmic::iced::widget::button(text(if app.is_dirty { "Save File *" } else { "Save File" }).font(font));
-        if app.selected_file.is_some() && app.is_dirty {
+        if app.selected_file.is_some() {
+          if app.is_dirty {
             cosmic::iced::widget::tooltip(
@@ -223,6 +223,15 @@
             )
             .into()
+          } else {
+            cosmic::iced::widget::tooltip(
+                btn,
+                "No unsaved changes.",
+                cosmic::iced::widget::tooltip::Position::Top,
+            )
+            .into()
+          }
         } else {
             cosmic::iced::widget::tooltip(
                 btn,
-                "No unsaved changes.",
+                "Select a file to enable saving.",
                 cosmic::iced::widget::tooltip::Position::Top,
